src/index.js

```diff
@@ -1,57 +1,31 @@
 import './index.css';
+import TaskList from './taskList.js';
 
-const allTasks = [
-  { description: 'complete project', completed: true, index: 1 },
-  { description: 'Learn webpack', completed: true, index: 2 },
-  { description: 'Reivew my code', completed: false, index: 3 },
-];
+const taskList = new TaskList();
+const taskListElement = document.getElementById('task-list');
+const addInput = document.getElementById('add');
+const editInput = document.getElementById('edit');
 
-const createTaskItem = (description) => {
-  const taskItem = document.createElement('li');
-  taskItem.className = 'task-item';
-
-  const checkbox = document.createElement('input');
-  checkbox.type = 'checkbox';
-  checkbox.className = 'form-check-input';
-
-  const taskDescription = document.createElement('p');
-  taskDescription.className = 'task-description';
-  taskDescription.textContent = description;
-
-  const deletebtn = document.createElement('button');
-  deletebtn.type = 'button';
-  deletebtn.className = 'delete-btn';
-  deletebtn.innerHTML = '<i class="bi bi-trash"></i>';
-
-  taskItem.appendChild(checkbox);
-  taskItem.appendChild(taskDescription);
-  taskItem.appendChild(deletebtn);
-
-  return taskItem;
-};
-
-const addTask = () => {
-  const inputField = document.querySelector('.add');
-  const description = inputField.value.trim();
+taskList.renderTasks(taskListElement);
 
+document.querySelector('.form-section').addEventListener('submit', (e) => {
+  e.preventDefault();
+  const description = addInput.value.trim();
   if (description !== '') {
-    const taskItem = createTaskItem(description);
-    const tasksList = document.getElementById('tasks-list');
-    tasksList.appendChild(taskItem);
-    inputField.value = '';
+    taskList.addTask(description);
+    taskList.renderTasks(taskListElement);
+    addInput.value = '';
   }
-};
-const renderTasks = () => {
-  const taskList = document.getElementById('task-list');
-
-  allTasks.forEach((task) => {
-    const taskItem = createTaskItem(task.description);
-    taskList.appendChild(taskItem);
-  });
-};
-window.addEventListener('DOMContentLoaded', () => {
-  renderTasks();
+});
 
-  const addButton = document.querySelector('.btn-add');
-  addButton.addEventListener('click', addTask);
-});
\ No newline at end of file
+document.querySelector('.edit-task-form').addEventListener('submit', (e) => {
```

Comment:
-  There is an error coming from here for briefly less than 1 second, as far as I can see is not affecting the overall project which I can still run and test it. But I would recommend you try to fix it